### PR TITLE
Allows PRs in test-infra to be automatically merged

### LINKED ISSUE #1762: [Prow auto-update] Config Tide to allow merging a PR without manual approval
Tide can be configured that allows PRs to be automatically merged without manual approval, see https://github.com/istio/test-infra/blob/6c79c4e28d0bc43f71142ce6cf6f96285f4fee1f/prow/config.yaml#L367 and the author field.

This is needed in [Prow auto-update](https://docs.google.com/document/d/1dziNlmNM3_VaPg46bgUV39K-PzBZAsmOUqfYQto99vg/edit#heading=h.q8uh8fpvt8m8) as we need to create a PR that copies staging-prow changes to production-prow, and we want this PR to be automatically merged rather than requiring manual approval.

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Allows PRs in test-infra to be automatically merged

**Which issue(s) this PR fixes**:
Fixes #1762

### CODE REVIEW

File: config/prow-staging/core/config.yaml [hunk 1]
```diff
@@ -108,11 +108,21 @@ tide:
...
     - do-not-merge/invalid-owners-file
     - needs-rebase
+  - repos:
+    # Allow PRs for test-infra to be automatically merged without manual approval.
+    - "knative-prow-robot/test-infra"
+    labels:
+    - "cla: yes"
+    - auto-merge
+    missingLabels: *knative_tide_missing_labels
+    author: knative-prow-robot
```
Comment (chaodaiG) [inline config/prow-staging/core/config.yaml]: I think it's not, but still want to double check, this is not the same bot creating auto images bump , right?

**Comment by chaodaiG** [comment]
/lgtm

/hold
feel free unhold if the answer to my question in the code is "no"

File: config/prow-staging/core/config.yaml (hunk 1 above)
Comment (chizhg) [reply config/prow-staging/core/config.yaml]: No, that bot's name is `knative-prow-updater-robot`.

**Pull request merged**